### ISSUE TITLE
Question with HER: AttributeError: 'Box' object has no attribute 'spaces'

### ISSUE THREAD

**Issue by yjc765** [hill-a/stable-baselines#417]
Hello,
I have a problem when I tried to use DDPG + HER.
The problem seems the definition of observation_space.

```bash
Traceback (most recent call last):
  File "/home/all-jy/git/jy_gym_stfl/train_ur5.py", line 35, in <module>
    policy_kwargs=dict(layers=[256, 256, 256]))
  File "/home/all-jy/.local/lib/python3.5/site-packages/stable_baselines/her/her.py", line 47, in __init__
    self._create_replay_wrapper(self.env)
  File "/home/all-jy/.local/lib/python3.5/site-packages/stable_baselines/her/her.py", line 62, in _create_replay_wrapper
    env = HERGoalEnvWrapper(env)
  File "/home/all-jy/.local/lib/python3.5/site-packages/stable_baselines/her/utils.py", line 25, in __init__
    self.spaces = list(env.observation_space.spaces.values())
AttributeError: 'Box' object has no attribute 'spaces'

```
The model is:
```python
model = HER('MlpPolicy', env, DDPG, n_sampled_goal=4,
            goal_selection_strategy='future',
            verbose=1, buffer_size=int(1e6),
            actor_lr=1e-3, critic_lr=1e-3,
            gamma=0.95, batch_size=256,
            policy_kwargs=dict(layers=[256, 256, 256]))
```
and observation_space I defined is:
```python
        self.action_space = spaces.Box(-np.inf, np.inf, shape=(7,), dtype='float32')
        obs = self._get_observation() # obs is a 1D array with 45 elements.
        self.observation_space = spaces.Box(-np.inf, np.inf, shape=obs.shape, dtype='float32')
```
![image](https://user-images.githubusercontent.com/33099087/61789554-4794ef80-ae15-11e9-9a3b-f011bf9f8269.png)
According to the error message, I need to add a attribute called 'space' to 'Box'? I'm confused.
Could anyone help me with this problem?

**Comment by araffin** [comment]
Hello, 
The answer is in the [documentation](https://stable-baselines.readthedocs.io/en/master/modules/her.html):
"
HER requires the environment to inherits from gym.GoalEnv
"

**Comment by PierreExeter** [comment]
Hello,

Thanks Araffin. How can I find out whether an environment inherits from gym.GoalEnv?

Obviously BitFlippingEnv does (as shown in the documentation) but most other Gym environments don't (well at least not the ones I tried: CartPole-v1, MountainCar-v0, Acrobot-v1, Pendulum-v0, MountainCarContinuous-v0)

**Comment by araffin** [comment]
>How can I find out whether an environment inherits from gym.GoalEnv?

Because gym does not have a proper documentation, you have to look at the source code.
In practice, only the `robotics` envs (FetchReach, PickAndPlace, ...) do, but you need a mujoco license for that.
The rule is `gym.Env`, `gym.GoalEnv` is the exception.

**Comment by kevin5k** [comment]
TLDR; 
"Difference in observation space type between gym.Env and gym.GoalEnv"

**Comment by MoritzNekolla** [comment]
Is it possible to write a wrapper or so which passes only a subset of that information into the SAC_Model? 

For example one wants to preprocess the input of the Sac algorithm.
